packages/server-wallet/src/evm-validator.ts

```diff
@@ -2,35 +2,42 @@ import {State, toNitroState} from '@statechannels/wallet-core';
 import {createValidTransitionTransaction} from '@statechannels/nitro-protocol';
 import * as PureEVM from 'pure-evm';
 import {utils} from 'ethers';
+import Objection from 'objection';
 
 import {AppBytecode} from './models/app-bytecode';
 import {logger} from './logger';
 import config from './config';
 
+const MISSING = '0x';
+const bytecodeCache: Record<string, string | undefined> = {};
+
 /**
  * Takes two states and runs the validateTransition in an evm (pureevm) if the bytecode exists in the DB.
  * Returns a promise that resolves to true if the validateTransition returns true false otherwise
  */
 export const validateTransitionWithEVM = async (
   from: State,
-  to: State
+  to: State,
+  tx: Objection.Transaction
 ): Promise<boolean | undefined> => {
   if (from.appDefinition !== to.appDefinition) {
-    throw new Error('States are using different appDefinitions');
+    logger.error('Invalid transition', {
+      error: new Error('States are using different appDefinitions'),
+    });
+    return false;
   }
-  const bytecode = await AppBytecode.getBytecode(
-    config.chainNetworkID,
-    from.appDefinition,
-    undefined
-  );
-  if (!bytecode) {
-    logger.warn(
-      `No bytecode found for appDefinition ${from.appDefinition} and chain id ${config.chainNetworkID}. Skipping valid transition check`
-    );
+  const bytecode =
+    bytecodeCache[from.appDefinition] ??
+    (bytecodeCache[from.appDefinition] =
+      (await AppBytecode.getBytecode(config.chainNetworkID, from.appDefinition, tx)) || MISSING);
+  if (bytecode === MISSING) {
+    // logger.warn(
+    //   `No bytecode found for appDefinition ${from.appDefinition} and chain id ${config.chainNetworkID}. Skipping valid transition check`
+    // );
```

Comment:
Are we just skipping this temporarily?

Reply:
At least temporarily: look at the output here: https://app.circleci.com/pipelines/github/statechannels/statechannels/8664/workflows/dd921a4a-c2ca-44f3-b57d-27398f57c48d/jobs/40820